### PR TITLE
Implemented Fyne receive page.

### PR DESCRIPTION
**Pull request by Sirmorrison**
image

Screenshot 2019-10-31 at 10 02 15 PM

Screenshot 2019-10-31 at 10 01 42 PM

image

image

### CODE REVIEW

File: fyne/pages/receive.go [hunk 1]
```diff
@@ -1,10 +1,287 @@
...
+		accountbalance := dcrutil.Amount(account.Balance.Total).String()
+		accountProperties.Append(widgets.NewHSpacer(84))
+		accountProperties.Append(widget.NewVBox(
+			widget.NewLabel(accountbalance),
+			spendableAmountLabel,
+		))
+
+		accountProperties.Append(widgets.NewHSpacer(8))
+
+		checkmarkIcon := widget.NewIcon(theme.ConfirmIcon())
```
Comment (Sirmorrison) [inline fyne/pages/receive.go]: from here.

**Comment by barisere** [comment]
@Sirmorrison please provide instructions on how to test the changes in this PR. I built the program and ran it, but I could not figure out how to get to the receive page. I got only the following screen.
Screen Shot fyne interface not implemented

**Review by barisere — CHANGES_REQUESTED** [review]
@Sirmorrison please provide instructions on how to test the changes in this PR. I built the program and ran it, but I could not figure out how to get to the receive page. I got only the following screen.
Screen Shot fyne interface not implemented

**Comment by Sirmorrison** [comment]
This is because you don't have any wallet account. You will need to create a wallet before you will be able to test this PR.

**Comment by barisere** [comment]
> This is because you don't have any wallet account. You will need to create a wallet before you will be able to test this PR.

No, I do. I had a wallet account before we changed direction on the fyne UI.
@Sirmorrison

**Comment by metaclips** [comment]
@barisere yes, it seems dcrlibwallet-wip branch and dcrlibwallet master branch detect/creates wallet in different ways

**Comment by Sirmorrison** [comment]
About the bug notice from the generate new address feature, It is a bug from dcrlibwallet  at it returns the initial wallet address and when generated again, returns a new address.
/Users/macbookair/Desktop/Screenshot 2019-10-24 at 6.19.35 PM.png/Users/macbookair/Desktop/Screenshot 2019-10-24 at 6.20.38 PM.png

**Comment by Sirmorrison** [comment]
About the bug notice from the generate new address feature, It is a bug from dcrlibwallet at it returns the initial wallet address and when generated again, returns a new address.

File: fyne/pages/receive.go [hunk 2]
```diff
@@ -1,10 +1,270 @@
...
+	"github.com/raedahgroup/dcrlibwallet"
+	"github.com/raedahgroup/godcr/fyne/assets"
+	"github.com/raedahgroup/godcr/fyne/widgets"
+	"github.com/skip2/go-qrcode"
 )
 
-func ReceivePageContent() fyne.CanvasObject {
-	return widget.NewLabelWithStyle("Receive", fyne.TextAlignLeading, fyne.TextStyle{Bold: true, Italic: true})
+const receivingDecredHint = "Each time you request a payment, a new \naddress is created to protect your privacy."
+const pageTitleText = "Receive DCR"
```
Comment (metaclips) [inline fyne/pages/receive.go]: Do we need this? receivingDecredHint and pageTitleText variables are only used in the ReceivePageContent function, there is really no use making it a package level variable.

File: fyne/pages/receive.go [hunk 3]
```diff
@@ -1,10 +1,270 @@
...
+
+var receiveHandler struct {
+	generatedReceiveAddress string
+	wallet                  *dcrlibwallet.LibWallet
+	generatedQrCode         []byte
+	accountNumber           uint32
+	selectedAccountName     string
+}
+
+func ReceivePageContent(dcrlw *dcrlibwallet.LibWallet, window fyne.Window, tabmenu *widget.TabContainer) fyne.CanvasObject {
```
Comment (metaclips) [inline fyne/pages/receive.go]: why are we declaring a struct outside a function and also passing function parameters in ReceivePageContent? only widgets that needs to be updated and refreshed by other functions should be defined outside the ReceivePageContent function.

File: fyne/pages/receive.go [hunk 4]
```diff
@@ -1,10 +1,270 @@
...
+	errorLabel := widget.NewLabelWithStyle("", fyne.TextAlignCenter, fyne.TextStyle{Bold: true})
+	errorLabel.Hide()
+	errorHandler := func(err string) {
+		errorLabel.SetText(err)
+		errorLabel.Show()
+	}
+
+	qrImage := widget.NewIcon(theme.FyneLogo())
+	generatedAddressLabel := widget.NewLabelWithStyle("", fyne.TextAlignLeading, fyne.TextStyle{Bold: true})
+	generateAddress := func(generateNewAddress bool) {
```
Comment (metaclips) [inline fyne/pages/receive.go]: Can we move this to generateAddressAndQrCode function?

File: fyne/pages/receive.go [hunk 5]
```diff
@@ -1,10 +1,270 @@
...
+			spendableLabel,
+		)
+
+		accountBalance := dcrutil.Amount(account.Balance.Total).String()
+		accountBalanceBox := widget.NewVBox(
+			widget.NewLabel(accountBalance),
+			spendableAmountLabel,
+		)
+
+		accountsView := widget.NewHBox(
```
Comment (metaclips) [inline fyne/pages/receive.go:126]: accountsView? can we be more explicit?

File: fyne/pages/receive.go [hunk 6]
```diff
@@ -1,10 +1,270 @@
...
+			widgets.NewHSpacer(15),
+			widget.NewIcon(icons[assets.ReceiveAccountIcon]),
+			widgets.NewHSpacer(15),
+			accountNameBox,
+			widgets.NewHSpacer(15),
+			accountBalanceBox,
+			widgets.NewHSpacer(15),
+		)
+
+		accountWidget := widgets.NewClickableBox(accountsView, func() {
```
Comment (metaclips) [inline fyne/pages/receive.go]: we should probably change the variable name to something better.

File: go.sum [hunk 7]
```diff
@@ -127,18 +127,21 @@ github.com/decred/dcrd/mining v1.1.0/go.mod h1:NQEtX604XgNwKcPFId1hVTTiBqmVQDlnq
 github.com/decred/dcrd/rpcclient v1.0.1/go.mod h1:tApXK3wwrAQtz7lcXeeqBwuktUZesvrFfvhAdedYqdM=
```
Comment (metaclips) [inline go.sum:119]: go mod tidy

**Comment by metaclips** [comment]
image

we should instead increase the height of the popup instead of the width

**Review by metaclips — CHANGES_REQUESTED** [review]
Spendable label text and amount color shouldn't be in black.

**Comment by metaclips** [comment]
Can we make popup width same as the account selector so as to allow uniformity?

**Comment by metaclips** [comment]
https://user-images.githubusercontent.com/27733432/67986619-bb704080-fc2b-11e9-9461-61c5b2fef594.png

We should right align balance label and left align account name and spendable label.

**Comment by metaclips** [comment]
image

We should right align balance label with the checkmark icon and left align account name and spendable label.

**Comment by Sirmorrison** [comment]
> image
> 
> We should right align balance label with the checkmark icon and left align account name and spendable label.

icons cannot be aligned left or right.

**Comment by metaclips** [comment]
Account dropdown content isn't correctly aligned, I think to do this we'll need to implement tables, this would be treated on a different PR.

**Review by metaclips — APPROVED** [review]
Tested locally and approved by me.

**Pull request merged**